### PR TITLE
feat: add function to handle local archive files

### PR DESCRIPTION
**Pull request by alethenorio**
Was originally removed because it is not used in this repo however it does not mean it is not used in other repos and is quite a useful function to have for working with local archive files for tools not yet introduced in this repo

### CODE REVIEW

File: sgtool/file.go [hunk 1]
```diff
@@ -55,6 +55,31 @@ func newFileState() *fileState {
 	}
 }
 
+func FromLocal(ctx context.Context, filepath string, opts ...Opt) error {
```
Comment (odsod) [inline sgtool/file.go:58]: Missing documentation here

Comment (odsod) [inline sgtool/file.go:58]: Missing documentation here

Comment (alethenorio) [reply sgtool/file.go:60]: Sure. Added some comments on the function

**Review by odsod — APPROVED** [review]
I would prefer to keep the API surface as small as possible, but if you say this is needed then let's go for it.

**Comment by alethenorio** [comment]
The main use case I have seen so far is for handling unsupported archive formats (such as bz2 though I think I may have a solution for that) where there is no native Go implementation so you end up using FromRemote to download the file, then some other way to decompress the file and then FromLocal to further process the resulting tar file.

**Review by viktorvoltaire — APPROVED** [review]
Nice :+1:

**Pull request merged**